std/traits.d

```diff
@@ -1482,6 +1486,193 @@ private FunctionAttribute extractAttribFlags(Attribs...)()
     return res;
 }
 
+/**
+Checks whether a function has the given attributes attached.
+
+Params:
+    func = function to check
+    attributes = variadic number of function attributes as strings
+
+Returns:
+    `true`, if the function has the list of attributes attached and `false` otherwise.
+
+See_Also:
+    $(LREF functionAttributes)
+*/
+template hasFunctionAttributes(args...)
+    if (isCallable!(args[0]) && args.length > 0
+         && allSatisfy!(isSomeString, typeof(args[1 .. $])))
+{
+    enum bool hasFunctionAttributes = {
+        import std.algorithm.searching : canFind;
+        import std.range : only;
+        enum funcAttribs = only(__traits(getFunctionAttributes, args[0]));
+        auto b = true;
+        foreach (attribute; args[1..$])
+        {
+            if (!funcAttribs.canFind(attribute))
+            {
+                b = false;
+                break;
+            }
+        }
+        return b;
+    }();
+}
+
+///
+unittest
+{
+    real func(real x) pure nothrow @safe
+    {
+        return x;
+    }
+    static assert(hasFunctionAttributes!(func, "@safe", "pure"));
+    static assert(!hasFunctionAttributes!(func, "@trusted"));
+
+    // for templates types are automatically inferred
+    bool myFunc(T)(T b)
+    {
+        return !b;
+    }
+    static assert(hasFunctionAttributes!(myFunc!bool, "@safe", "pure", "@nogc", "nothrow"));
+    static assert(!hasFunctionAttributes!(myFunc!bool, "@trusted"));
+}
+
+unittest
+{
+    struct S
+    {
+        int noF() { return 0; }
+        int constF() const { return 0; }
+        int immutableF() immutable { return 0; }
+        int inoutF() inout { return 0; }
+        int sharedF() shared { return 0; }
+
+        int x;
+        ref int refF() return { return x; }
+        int propertyF() @property { return 0; }
+        int nothrowF() nothrow { return 0; }
+        int nogcF() @nogc { return 0; }
+
+        int systemF() @system { return 0; }
+        int trustedF() @trusted { return 0; }
+        int safeF() @safe { return 0; }
+
+        int pureF() pure { return 0; }
+    }
```

Comment:
CodeCov is failing because DMD counts  all these functions as uncovered.
(there's a [NG thread](http://forum.dlang.org/post/nncm00$2sc8$1@digitalmars.com) discussion about not counting unittest blocks, but tl:dr it won't happen)